scipy/stats/_kde.py

```diff
@@ -636,6 +636,155 @@ def logpdf(self, x):
+            msg = "axis have dimension %s, dataset has dimension %s" % \
+                (ax.size, self.d)
+            raise ValueError(msg)
+
+        if d != ax.size:
+            if d == 1 and m == ax.size:
+                # points was passed in as a row vector
+                points = reshape(points, (m, 1))
+                d = ax.size
+                m = 1
+            else:
+                msg = "points have dimension %s, axis has dimension %s" % \
+                    (d, ax.size)
+
+        cov = self.covariance[ax[:, newaxis], ax]
+        dataset = self.dataset[ax]
+        diff = dataset[:, :, np.newaxis] - points[:, np.newaxis, :]
+        # (# of dim, # of data, # of points)
+
+        if d == 1:
+            norm_factor = sqrt(2 * pi * cov[0, 0])
+            icd = diff / cov
+        else:
+            norm_factor = (2 * pi) ** (d / 2) * linalg.det(cov) ** (1 / 2)
+            icd = linalg.solve(cov, diff, assume_a='pos')
+
+        energy = sum(diff * icd, axis=0) / 2
+        result = (self.weights @ np.exp(-energy)) / norm_factor
+
+        return result
+
+    def logpdf_marginal(self, x, axis):
+        """Evaluate the log of estimated marginal pdf on a provided set of points.
+
+        Notes
+        -----
+        .. versionadded:: 1.3.0
+
+        See the ``pdf_marginal`` docstring for more details.
+        """
+        points = atleast_2d(x)
+        ax = atleast_1d(axis)
+
+        if not np.issubdtype(ax.dtype, np.integer):
+            msg = "elements of axis should be intergers"
+            raise ValueError(msg)
+
+        d, m = points.shape
+        if ax.size > self.d or ax.size == 0:
+            msg = "axis have dimension %s, dataset has dimension %s" % \
+                (ax.size, self.d)
+            raise ValueError(msg)
+
+        if d != ax.size:
+            if d == 1 and m == ax.size:
+                # points was passed in as a row vector
+                points = reshape(points, (m, 1))
+                d = ax.size
+                m = 1
+            else:
+                msg = "points have dimension %s, axis has dimension %s" % \
+                    (d, ax.size)
+
+        cov = self.covariance[ax[:, newaxis], ax]
+        dataset = self.dataset[ax]
+        diff = dataset[:, :, np.newaxis] - points[:, np.newaxis, :]
+        # (# of dim, # of data, # of points)
+
+        if d == 1:
+            norm_factor = sqrt(2 * pi * cov[0, 0])
+            icd = diff / cov
+        else:
+            norm_factor = (2 * pi) ** (d / 2) * linalg.det(cov) ** (1 / 2)
+            icd = linalg.solve(cov, diff, assume_a='pos')
+
+        energy = sum(diff * icd, axis=0) / 2
+        result = logsumexp(-energy.T, b=self.weights / norm_factor, axis=1)
+
+        return result
+
+    def marginalize(self, axis):
+        """Generate a marginal KDE distribution
+
+        Parameters
+        ----------
+        axis : int or 1-d array_like
+            Axis (axes) along which the margins are to be computed.
```

Comment:
These are subject to discussion. See [questions](https://github.com/scipy/scipy/pull/9932#issuecomment-1211375216) about function name, parameter name, and parameter meaning.

Reply:
I would maybe go with `d`, `dimensions` or simply `marginals`. I also agree with you that this parameter should represent the dimensions we want to keep.